crates/conf/src/plugin.rs

```diff
@@ -25,16 +28,23 @@ impl Plugin for ConfPlugin {
 }
 
 #[derive(Resource)]
-struct LoadingTask(Task<Result<Configuration>>);
+struct LoadingTask(Task<Result<Configuration, ConfigLoadError>>);
 
 fn cleanup(mut commands: Commands) {
     commands.remove_resource::<LoadingTask>();
 }
 
 fn start_loading(mut commands: Commands) {
     let task = IoTaskPool::get().spawn(async {
-        let path = conf_dir()?.join("conf.yaml");
-        load_conf(path.as_path()).await
+        let Some(base_conf_dir) = dirs::config_dir() else {
```

Comment:
What is the reason for this change? 

Note that `thiserror` Crate has `#[from]` attribute which we can utilize to convert errors from `conf_dir()`. I think it is desirable to utilize functions from `de_core::fs` to stay DRY and keep it organized. A variant of the following might suffice:

```rust
let path = conf_dir().map_err(ConfigLoadError::from)?.join("conf.yaml");
Configuration::load(path.as_path()).await
```